pandas/tests/series/test_constructors.py

```diff
@@ -909,6 +909,11 @@ def test_constructor_with_naive_string_and_datetimetz_dtype(self, arg):
         expected = Series(pd.Timestamp(arg)).dt.tz_localize('CET')
         assert_series_equal(result, expected)
 
+    def test_constructor_datetime64_outofbound(self):
+        # GH-26206 out of bound non-ns unit
+        with pytest.raises(pd.errors.OutOfBoundsDatetime):
+            pd.Series(np.array(['2262-04-12'], dtype='datetime64[D]'))
```

Comment:
Could you also test the minimum date as well?

Reply:
It's relying on a general routine that does this bound checking and which is already tested thoroughly through other means, here I mainly want to test to ensure this routine is used. Therefore, I decided to keep the tests here a bit simpler (adding is not that hard, but it just complicates the test further, and there are already several cases). Is that OK?